rir/src/interpreter/runtime.cpp

```diff
@@ -28,27 +28,45 @@ Function* isValidClosureSEXP(SEXP closure) {
 
 // for now, we will have to rewrite this when it goes to GNU-R proper
 
-void printFunction(Function* f) {
-    Rprintf("Function object (%p):\n", f);
-    Rprintf("  Magic:           %x (hex)\n", f->info.magic);
-    Rprintf("  Size:            %u\n", f->size);
-    Rprintf("  Origin:          %p %s\n", f->origin(), f->origin() ? "" : "(unoptimized)");
-    Rprintf("  Next:            %p\n", f->next());
-    Rprintf("  Code objects:    %u\n", f->codeLength);
-    Rprintf("  Fun code addr:   %x (hex)\n", f->body());
-    Rprintf("  Invoked:         %u\n", f->invocationCount);
-    Rprintf("  Signature:       %p\n", f->signature);
-    if (f->signature)
-        f->signature->print();
+void printFunction(Function* f, std::ostream& out) {
```

Comment:
This actually looks uglier, but the advantage is that we can control where the printing happens, rather than trying to hack Rprintf.